### ISSUE TITLE
Collapsible Side Panel

### ISSUE THREAD

**Issue by SheepDomination** [aminought/firefox-second-sidebar#46]
Can you please make it so that it's collapsible.

**Comment by mikegrogan** [comment]
Just installed this. Coming from Vivaldi this is great so far, but also was wondering about making it collapsible. Thanks for the hard work.

**Comment by VlaK0r** [comment]
Great idea!

**Comment by aminought** [comment]
Now you can enable auto-hiding in sidebar settings

**Comment by guigirl42** [comment]
> auto-hiding in sidebar settings

Oh wow, this is so great. Your project just gets better & better. Many thanks!

**Comment by aminought** [comment]
@guigirl42 Don't panic, there hasn't even been an official release yet. :)

I don't know why this problem occurs because I usually test it on Stable and NIghtly for Ubuntu and Windows. And I can't reproduce this bug, the Second Sidebar works perfectly for me in two devices.

> when closing eventually succeeds, the confirmation dialogue asks if i wish to close TWO windows [despite me only having one open]
That's ok, look here: https://github.com/aminought/firefox-second-sidebar/issues/72#issuecomment-2683059428

**Comment by guigirl42** [comment]
> Don't panic, there hasn't even been an official release yet. :)

Yes indeed. It's my own fault i know, for grabbing the current WIP version [`Commits on Feb 26, 2025`] instead of patiently waiting for release of the official version. I only posted in case you might not have known of the problem. 

> That's ok, look here: https://github.com/aminought/firefox-second-sidebar/issues/72#issuecomment-2683059428

Ok but only after installing the new files per `Commits on Feb 26, 2025` do restarts present the almost empty window, with the "real" one needing me to restore from `History - Recently Closed Windows`. Reverting to your current last "official" release version 1.3 of Jan5, restores the normal window behaviour at close & relaunch. 

Anyway, i have learned my lesson; i shall ignore all your WIP updates, & not update my files til your next _official_ version. :-)

**Comment by aminought** [comment]
Do not ignore, please, I need feedback because I have to fix all bugs before release :)

**Comment by aminought** [comment]
> Anyway, i have learned my lesson; i shall ignore all your WIP updates, & not update my files til your next _official_ version. :-)

That was my answer to this. But if you want to help with the current case, you can record your screen and show me the occurrence of this error.

**Comment by SheepDomination** [comment]
The upcoming version will be collapsible if the option is set, as well as the ability to pin; either way ?
One other request is to allow favorite web sites to the window.

**Comment by guigirl42** [comment]
> you can record your screen

If you mean a video, then no, i am unwilling to post that, but instead i will post static screenshots i have edited. 

![Image](https://github.com/user-attachments/assets/2c9c6218-819f-4d0e-a281-7f89b5fb418c)

![Image](https://github.com/user-attachments/assets/ad15ffe3-db14-4499-9586-ba848d8f593b) 

Ahhhhh, so far, tonight, in this re-test with FF, I have _not_ experienced _any_ recurrences of the previous problem of the window which opens after the restart losing all but one of the open tabs, with all the rest being in a second window needing recovering from the Closed Windows History. I am surprised, as atm i cannot think of anything i have done differently tonight, compared to the other night. 

The only ongoing problem with this latest commit is the _impossibility to close the window when i want_ to... tonight it took even **20 & 30 attempts** before it worked. 

Later tonight i shall repeat this retest, in my Nightly profile also with Sidebery. I shall update this post, or reply to it, if anything significant occurs.

**Comment by aminought** [comment]
Thank you! Can you please open [Browser Toolbox](https://firefox-source-docs.mozilla.org/devtools-user/browser_toolbox/index.html) and check if some logs appear when you try to close a window?

**Comment by guigirl42** [comment]
I've not yet had time to try the `Toolbox`, because when i relaunched FF to try what you asked, the "window problem" recurred! 

![Image](https://github.com/user-attachments/assets/91917146-35ba-423b-b236-ccfc2cdf791c)

![Image](https://github.com/user-attachments/assets/63752985-fddd-4543-822c-fb6baca4458f)

**Comment by guigirl42** [comment]
> not yet had time to try the `Toolbox`

Now i _have_ tried the `Toolbox`, but there were no errors... BECAUSE... all seven (7) times that i launched FF this time, it opened with the single-tab "wrong" window, & i had to then recover the "real" window from History. Seven out of seven times when i then closed these windows, first the "wrong" one, then the "real" one, the close operation worked correctly. 

I don't yet have enough data to _prove_ this, but from memory i _think_ that every time i could not easily close the window _only_ happened when the preceding FF launch had _correctly opened only the "real" window_. Otoh, it seems that on each occasion when the "wrong" window opened, necessitating me manually recovering the "real" window, Closing them works correctly. 

Does this make any sense to you?

**Comment by SheepDomination** [comment]
This new collapsible side bar, will it have an option to auto-collapse ?

**Comment by aminought** [comment]
> This new collapsible side bar, will it have an option to auto-collapse ?

It has already.

**Comment by SheepDomination** [comment]
> > This new collapsible side bar, will it have an option to auto-collapse ?
> 
> It has already.

It is ready the update ?

**Comment by aminought** [comment]
> It is ready the update ?

You can try :)
Actually, I need some early adopters to test new features before release, so you can install the version from the master branch and create an issue if you notice any problems.

**Comment by guigirl42** [comment]
> I need some early adopters to test new features before release, so you can install the version from the master branch and create an issue if you notice any problems

Well, i already did test it, & posted my unpleasant results above, which you said you could not reproduce, so now i am confused what to do.

**Comment by aminought** [comment]
> Well, i already did test it, & posted my unpleasant results above, which you said you could not reproduce, so now i am confused what to do.

Your problem is not relevant to this issue and I've fixed it already.
